compiler/rustc_mir/src/transform/large_enums.rs

```diff
@@ -0,0 +1,276 @@
+                                    )]),
+                                })),
+                            )),
+                        };
+
+                        let dst =
+                            Place::from(local_decls.push(LocalDecl::new(tcx.mk_mut_ptr(ty), span)));
+
+                        let dst_ptr = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (
+                                dst,
+                                Rvalue::AddressOf(Mutability::Mut, *lhs),
+                            )),
+                        };
+
+                        let dst_cast_ty = tcx.mk_mut_ptr(tcx.types.u8);
+                        let dst_cast_place =
+                            Place::from(local_decls.push(LocalDecl::new(dst_cast_ty, span)));
+
+                        let dst_cast = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (
+                                dst_cast_place,
+                                Rvalue::Cast(CastKind::Misc, Operand::Copy(dst), dst_cast_ty),
+                            )),
+                        };
+
+                        let src =
+                            Place::from(local_decls.push(LocalDecl::new(tcx.mk_imm_ptr(ty), span)));
+
+                        let src_ptr = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (
+                                src,
+                                Rvalue::AddressOf(Mutability::Not, *rhs),
+                            )),
+                        };
+
+                        let src_cast_ty = tcx.mk_imm_ptr(tcx.types.u8);
+                        let src_cast_place =
+                            Place::from(local_decls.push(LocalDecl::new(src_cast_ty, span)));
+
+                        let src_cast = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (
+                                src_cast_place,
+                                Rvalue::Cast(CastKind::Misc, Operand::Copy(src), src_cast_ty),
+                            )),
+                        };
+
+                        let copy_bytes = Statement {
+                            source_info,
+                            kind: StatementKind::CopyNonOverlapping(box CopyNonOverlapping {
+                                src: Operand::Copy(src_cast_place),
+                                dst: Operand::Copy(dst_cast_place),
+                                count: Operand::Constant(
+                                    box (Constant {
+                                        span,
+                                        user_ty: None,
+                                        literal: ConstantKind::Val(
+                                            interpret::ConstValue::from_u64(total_size.bytes()),
```

Comment:
It should be `from_machine_usize` to produce a constant with correct size on non-64 bit targets. At this point we could also use variant size calculated earlier.